src/rgw/rgw_common.cc

```diff
@@ -874,7 +874,17 @@ int RGWHTTPArgs::parse(const DoutPrefixProvider *dpp)
         });
       }
       string& val = nv.get_val();
-      ldpp_dout(dpp, 10) << "name: " << name << " val: " << val << dendl;
+      static constexpr std::initializer_list<const char*>
+          sensitive_keyword_list = {"password"};
+      bool is_sensitive = false;
+      for (const auto& key : sensitive_keyword_list) {
+        if (name.find(key) != std::string::npos) {
+          is_sensitive = true;
+          break;
+        }
+      }
+      ldpp_dout(dpp, 10) << "name: " << name
+                         << " val: " << (is_sensitive ? "****" : val) << dendl;
```

Comment:
isn't this information readily available from commands like `radosgw-admin topic get` and `metadata get`? do we expect these logs to be exposed to someone that doesn't already have admin access?

Reply:
i think there are cases for that.
if someone hooks our debug logs to syslog, or in case of rook, they may be shipped outside of the actual rgw machine.
agree that the demarcation between sensitive and other info in logs is a bit tricky. in many places usernames and hostnames are also considered sensitive info

Reply:
@cbodley, if this comment is resolved can we merge this PR ?